### DISCUSSION TITLE
An IntelliJ-based plugin

### DISCUSSION THREAD

**Discussion by benoitjadinon** [cline/cline#581] [Feature Requests]
I'm using Cline in VSCode, then switching back and forth to WebStorm for whatever manual fine-tuning is left to do.
This request will hopefully replace #63 (that has gathered quite some support already)

**Comment by benoitjadinon** [comment]
did a search on "intelli" but not on "jetbrains"

**Reply by benoitjadinon** [reply]
#501

**Comment by doiyuki** [comment]
I want this feature too

**Comment by jwangkun** [comment]
I want this feature too

**Comment by flyer88** [comment]
I want this feature too

**Comment by ngby** [comment]
I want this feature too

**Comment by dcbartlett** [comment]
I've moved the conversations from https://github.com/cline/cline/discussions/501 and https://github.com/cline/cline/discussions/1242 to here.  Please continue any relevant conversions regarding this topic.  We have seen a significant rise in the want for a JetBrains plugin and are actively working to field the idea.

**Reply by Nopik** [reply]
Do you have any rough ETA for it?

**Reply by emaggini-1** [reply]
![image](https://github.com/user-attachments/assets/88df1290-24e2-430c-9423-8e21ae86f566)

**Reply by mmihalev** [reply]
Any news on this?

**Reply by oxygenecore** [reply]
Are we going to have anything here, or did cline just deepseeked Junie?

**Comment by Joseph-Cardwell** [comment]
it has to run in a java wrapper right?

**Reply by KaKi87** [reply]
Who are you to downvote all enthusiastic responses toward a JetBrains integration ?

**Comment by africa1207** [comment]
I want this feature too

**Comment by LousyBook-01** [comment]
hope this happens soon

**Comment by michabbb** [comment]
and again, i have to unsubscribe because of all these morons who are unable to hit the ⬆️ button and instead have nothing better to do than annoy everyone with these super intelligent "I want this too" spam postings 🙄 🤦 WTF.....

**Reply by Bratzmeister** [reply]
+1 on what this guy said

**Reply by huyz** [reply]
All these comments should be 👎 by everyone else until folks GitHub-wide get the message.

**Reply by huyz** [reply]
Since morons will stay morons, I resorted to creating a custom Gmail filter to trash these email notifications (regular Gmail filters can't see symbols like `+`).
Ask ChatGPT how to create a Google Apps Script for Gmail and have it contain:

```js
function filterClinePlusOneEmails() {
  var threads = GmailApp.search('list:cline.cline.github.com subject:"An IntelliJ-based plugin" newer_than:1d'); // Search for relevant emails
  for (var i = 0; i < threads.length; i++) {
    var messages = threads[i].getMessages();
    for (var j = 0; j < messages.length; j++) {
      var message = messages[j];
      var body = message.getPlainBody();

      if (body.includes("+1") || body.includes("I want this feature")) {
        message.moveToTrash(); 
        Logger.log("Trashed email: " + message.getSubject());
      }
    }
  }
}
```

If you have a Google Workspace account, great, as you can configure it to run in real-time. Otherwise, for a free personal account, the minimum interval is every minute.

**Reply by LivioGama** [reply]
How about using ChatGPT or Cline to build the JetBrains plugin rather than Gmail filters :D ?

**Reply by huyz** [reply]
@LivioGama go for it :) let us know how that goes

**Comment by JamesEdward-X** [comment]
when can I use Cline in JetBrains IDE

**Comment by guiant2023** [comment]
tambem quero!

**Comment by HiromiShikata** [comment]
Cline is waiting team leader for this, you should apply :eyes:
https://github.com/cline/cline/discussions/1242#discussioncomment-11801771

**Comment by vgpastor** [comment]
Hi!

While we waiting to the team leader, i think that can be a great idea, speak about the refactor needs to improve scalability.

I open this discussion to speak abut this https://github.com/cline/cline/discussions/1388

**Comment by x1064991287** [comment]
I want this feature too

**Comment by johnwbaxter** [comment]
+1 please!

**Comment by anjiava** [comment]
+1 please!

**Comment by ofirgeller** [comment]
you are asking for another product, will draw too much effort from cline for vsc.

**Comment by RatmmmhSquishyRat** [comment]
i dont want this feature yet, I use both editors. The plugin still need many improvements itself.

**Comment by andydavid-dev** [comment]
+1 please!

**Comment by Daryl2023** [comment]
I want this feature too

**Comment by brightkut** [comment]
+1

**Comment by maichaojiu** [comment]
+1 please!

**Comment by hey-jude** [comment]
+1 please!

**Comment by sylvanSuen** [comment]
+1 please!

**Comment by chathurabuddi** [comment]
+1

**Comment by wzlstudy** [comment]
I want this feature too

**Comment by Dlinw** [comment]
I want this feature too
+999999999

**Comment by Tcheutchoua-Steve** [comment]
+1 for this feature

**Comment by sean-redteam** [comment]
+1

**Comment by aliounembaye2211** [comment]
I want this feature

**Comment by DevRyuki** [comment]
+1

**Comment by ishii-h-a** [comment]
+1

**Comment by lyousan** [comment]
+1 please!

**Comment by laiso** [comment]
Has anyone designed this yet? We could use Cline for this task. 
The plugin would likely be written mostly in Kotlin and swing. It might be possible to embed it using WebView, but the parts of Cline that include `import VSCode` wouldn’t be usable internally. So, I think it’d have to be built from scratch.

**Reply by laiso** [reply]
Combining it with command-line tools like [aider](https://aider.chat/), [Goose](https://github.com/block/goose), or [Claude Code](https://docs.anthropic.com/en/docs/agents-and-tools/claude-code/overview) could be interesting. For example, Goose has a guide for fetching IntelliJ-based editor information through the MCP plugin. 
https://block.github.io/goose/docs/tutorials/jetbrains-mcp/
However, I personally prefer the visual representation that Cline offers, so having an alternative tool like this would be a nice option. Unfortunately, [Junie](https://www.jetbrains.com/junie/) isn’t enabled for me yet.

**Comment by Jason5Lee** [comment]
I am looking for an IntelliJ-based plugin for AI programming that supports custom API configuration. This would allow me to use my own OpenAI or OpenRouter API key, rather than being locked into a specific subscription. For VSCode, CLINE is my go-to option.

**Reply by wzlstudy** [reply]
[ProxyAI](https://plugins.jetbrains.com/plugin/21056-proxy-ai) can help you

**Reply by alp82** [reply]
Also [Codeium](https://plugins.jetbrains.com/plugin/20540-codeium-ai-autocomplete-and-chat-for-python-js-java-go--)

**Reply by thibaud57** [reply]
ProxyAi looks cool thanks

**Comment by JamesEdward-X** [comment]
why hasn't Cline plugin been listed on the JetBrains IDE plugin marketplace yet? It's already been four months

**Reply by huyz** [reply]
> why hasn't Cline plugin been listed on the JetBrains IDE plugin marketplace yet? It's already been four months

Because you can't just list it. It has to be reimplemented entirely

**Comment by kxkw** [comment]
I want this feature too

**Comment by brockbutler** [comment]
+1

**Comment by nikhil-zlai** [comment]
+1 - willing to pay for it 🙏

**Comment by patrickxunuo** [comment]
+1

**Comment by techjewel** [comment]
+1
willing to pay for it 🙏

**Reply by Keshaun1222** [reply]
Same, ngl

**Comment by siddg97** [comment]
+1

**Comment by paarai** [comment]
I want this feature too

**Comment by felka** [comment]
+1

**Comment by emedina** [comment]
+1

**Comment by zhsama** [comment]
+1

**Comment by xyh67** [comment]
+1

**Comment by Constantilation** [comment]
+1

**Comment by GaelleChapuis** [comment]
+1

**Comment by RazgrizHsu** [comment]
+1, please

**Comment by AspireOne** [comment]
I want this feature too

**Comment by FlavioSantoro92** [comment]
+1

**Comment by Bohemicus** [comment]
I want this feature too

**Comment by anfreshman** [comment]
+1

**Reply by alp82** [reply]
Reminder to folks that there's an upvote button on the original description. Use this, rather than adding more & more "+1" comments over & over, since it just creates a lot of unnecessary noise that gets in the way of the conversation and is likely to actually slow down development on this.

**Comment by tyaqing** [comment]
+1

**Reply by alp82** [reply]
Reminder to folks that there's an upvote button on the original description. Use this, rather than adding more & more "+1" comments over & over, since it just creates a lot of unnecessary noise that gets in the way of the conversation and is likely to actually slow down development on this.

**Comment by toby-griffiths** [comment]
Reminder to folks that there's an upvote button on the original description.   Use this, rather than adding more & more "+1" comments over & over, since it just creates a lot of unnecessary noise that gets in the way of the conversation and is likely to actually slow down development on this.

**Comment by halleyxu** [comment]
+1

**Reply by angelxmoreno** [reply]
> Reminder to folks that there's an upvote button on the original description.   Use this, rather than adding more & more "+1" comments over & over, since it just creates a lot of unnecessary noise that gets in the way of the conversation and is likely to actually slow down development on this.

**Comment by GameCodeMode** [comment]
its been almost 6 months, so is there any update?

**Comment by wboult** [comment]
I tried JetBrains' coding agent [Junie](https://plugins.jetbrains.com/plugin/26104-jetbrains-junie) as part of the free trial. Thought I'd share my experience since I didn't see any mention of Junie here, hope it doesn't deflect from the purpose of this discussion too much and is useful (if not I'll delete). I have no affiliation with Jetbrains, just use Cline and Intellij a lot so was interested to compare.

![junie](https://github.com/user-attachments/assets/a8c62d3c-35d2-4aa1-b20a-fea36b4845d9)

**Pros**
- Really polished UX — feels slick and well-integrated into IntelliJ.
- Nice not having to jump between VSCode and IntelliJ to use Cline.

**Cons**
- Doesn't yet support customising the LLMs used. There's [an issue raised](https://youtrack.jetbrains.com/issue/LLM-11585/Bring-your-own-LLM-API-key-for-individual-personal-license-of-AI-Pro) so they will probably address this at some point. As far as I can tell it uses a mix of LLMs based on the task.
- Quality and speed felt behind Cline + Claude/GPT/Gemini. Didn't time it but felt like it was taking ~2x the time I'd expect Cline to for medium-complexity tasks.
- Limited configurability so far, there's a "project guidelines" file equivalent to `.clinerules` or custom instructions. It's early days and Jetbrains' do make great products, so I expect it won't be long for more of this stuff to start being added.

JetBrains could eventually have an edge on tight IDE integration — especially around build system metadata, debugging, and run configurations. It would be pretty cool to have the agent launching the debugger, inspecting live state, and integrating these sources of information to start doing more complicated programming tasks, enriching the context and improving accuracy and efficiency. There’s [some mention of this direction](https://medium.com/h7w/jetbrains-just-dropped-junie-meet-their-new-ai-can-beat-github-copilot-in-vs-code-1bcbc15e6f1f) already. It's harder to do this in VSCode where these types of integrations aren't as standardised. 

Still feels early and for me doesn't feel that competitive with Cline/Roo yet, but interesting one to watch with Jetbrains behind it. Hopefully they have plans to make an open-source community edition too.

**Reply by cliffordp** [reply]
I just tried PhpStorm's AI chat while editing JavaScript and it feels nice but stinks overall. It might be less efficient and effective than copy-paste into any AI website. I installed VS Code just to try Cline and in 5 minutes was happier than the days of using PhpStorm's AI chat. I will say that PhpStorm's AI autocompletion suggestions are nice to have but added a syntax error at the end every time I hit Tab (like not removing the closing parthesis or something). Also, PhpStorm's AI wanted to enforce its naming conventions instead of follow my existing ones even to the point of creating errors for undefined variables.

I will say that I burned through Cline credits way more quickly than expected, but to get my multiple problems solved within an hour was worth it.

**Comment by richard-collette-precisely** [comment]
Can CLINE create its own JetBrains plugin?

**Comment by alp82** [comment]
Windsurf released a plugin for Jetbrains and it works incredibly well. Everyone looking for an alternative should check out the free tier.

**Reply by steveepreston** [reply]
Sadly they disabled custom api endpoint, so you can't edit the api endpoint

**Comment by qianxueyiran** [comment]
I want this feature too

**Reply by angelxmoreno** [reply]
Reminder to folks that there's an upvote button on the original description. Use this, rather than adding more & more "+1" comments over & over, since it just creates a lot of unnecessary noise that gets in the way of the conversation and is likely to actually slow down development on this.

**Reply by steveepreston** [reply]
@angelxmoreno hmmm, are you sure the notification slow down the dev?

**Comment by holyhlifuga** [comment]
I want this feature too

**Reply by angelxmoreno** [reply]
Reminder to folks that there's an upvote button on the original description. Use this, rather than adding more & more "+1" comments over & over, since it just creates a lot of unnecessary noise that gets in the way of the conversation and is likely to actually slow down development on this.

**Comment by yssa2017wym** [comment]
when can I use Cline in JetBrains IDE

**Comment by Prikalel** [comment]
I want this feature too

**Reply by angelxmoreno** [reply]
Reminder to folks that there's an upvote button on the original description. Use this, rather than adding more & more "+1" comments over & over, since it just creates a lot of unnecessary noise that gets in the way of the conversation and is likely to actually slow down development on this.

**Comment by yshavit-tulip** [comment]
@benoitjadinon Would it be possible to [lock this discussion][1], so that only you (and others with write access to the repo) can respond? That would let people who want updates to this subscribe to it, while removing the "+1 me too" noise.

[1]: https://docs.github.com/en/communities/moderating-comments-and-conversations/locking-conversations

**Reply by benoitjadinon** [reply]
my bad, but how do I do that without admin rights ? and why don't admins do it ?

**Reply by yshavit-tulip** [reply]
Oops, my bad! I saw the `Author` tag and misread it as `Maintainer` :( Yeah, it would have to be the admins.

@dcbartlett any chance we could get this discussion locked down to only project maintainers?

**Comment by baoxin** [comment]
![image](https://github.com/user-attachments/assets/baa11056-c5f5-48ba-8c73-14f66dd729ab)
What about these people's brains?

**Reply by alp82** [reply]
they have a functioning one because they understand that adding useless comments is wasteful and creates unnecessary notifications for every person who is subscribed. don't be on the other side.

**Comment by kevinneung** [comment]
Hi all! Thanks for all of the excitement and enthusiasm. I wanted to stop by and give an update. The team has been working diligently to get a Jetbrains integration launched. Lots of progress has been made, so stay tuned!

**Reply by ghostrider0470** [reply]
Thanks so much, this feature will be amazing

**Reply by vgpastor** [reply]
It's possible to collaborate in a preview version to provide feedback?

**Reply by RomanBrodetski** [reply]
very cool, looking forward!

**Reply by timkindberg** [reply]
Thank you!!

**Reply by dshumsky** [reply]
Thank you!!!

**Reply by steveepreston** [reply]
Great ❤️
We are still waiting for `Cline`

**Reply by jclab-joseph** [reply]
Thank you so much!

**Reply by Akshay314** [reply]
Hi, Is there any approximate ETA? Anxiously waiting since a long time!

**Reply by ruchernchong** [reply]
Awesome! Looking forward to it!

**Reply by 8r4n** [reply]
ETA please.

**Comment by lovechier** [comment]
I hope that CLINE will be applied to inetllij as soon as possible

**Comment by neo7** [comment]
This is almost the last week of June-2025, I am hoping to get some exciting news 😃

**Reply by leonzy** [reply]
Is there any progress?

**Reply by febinct** [reply]
any expected time line

**Comment by Akshay314** [comment]
I think I am almost adapted to VS Code now while waiting for this plugin eternally for IntelliJ. Its gonna loose too many customers just because of the delays in GenAI adoption. World is changing, they didn't.

**Reply by KaKi87** [reply]
Not true.

JetBrains might be slower (honestly I don't even feel like they are slower), but it doesn't matter, because their customers are also companies that move slow anyway, especially in terms of allowing third-parties to access their source code.

With that said :

- JetBrains had a cloud LLM in the IDE 7 months before CLine was first released (in public beta 13 months before CLine was released) ;
- JetBrains had a local LLM in the IDE 3 months before CLine was first released ;
- JetBrains got Ollama support in the IDE 2 months after CLine did.

**Reply by steveepreston** [reply]
Agree with @KaKi87
Also you can check `AI Coding` and `Proxy AI` in JetBrains store, they have some features

**Reply by dredivaris** [reply]
You also might want to try Augment Code - it's available both for VSCode and Jetbrains.  Its contextual understanding of large codebases is pretty impressive.

**Reply by KaKi87** [reply]
> Augment Code - it's available both for VSCode and Jetbrains

No Ollama support though.

**Reply by dredivaris** [reply]
True, it obfuscates the model (though it's currently mostly using Claude 4 Sonnet)

**Reply by KaKi87** [reply]
Also, I agree with this : [Victor Savkin: How JetBrains Is Similar to Apple | The WebStorm Blog](https://blog.jetbrains.com/webstorm/2021/06/victor-savkin-how-jetbrains-is-similar-to-apple/)

> I sometimes joke that JetBrains is a lot like Apple.
>
> **What do you mean when you say that JetBrains is like Apple?**
>
> It may take a bit longer for certain features to arrive, but once they do they tend to be well-integrated and of better quality.

I hate Apple though. But I must acknowledge the quality.

**Comment by richasinha** [comment]
Please enable this feature on IntelliJ

**Comment by l3est** [comment]
This is the top feature request, with ten times more votes than the next one, but it still hasn't been done.

**Comment by adaltonreis** [comment]
I am also looking forward to that. Cline is by far the best extension ( using Claude 4). I tried other options with the same AI model (Claude 4), including the AI assistance from Jet Brains, but no doubt, Cline is unbeatable!

It would really be great to be able to use it in the InteliJ IDE

**Comment by KaKi87** [comment]
@yshavit @yshavit-tulip wow, using two accounts to downvote now, that's next level shit !

![](https://github.com/user-attachments/assets/fc41c8b6-9787-4f6a-a0c3-a98656ecaa9a)

Why though ?

**Reply by yshavit** [reply]
Heh, only meant to downvote for one. I'll retract the other.

The reason for the downvote is that you're sending everyone who's following this issue a message saying you're also interested in it, rather than just upvoting the issue.

**Reply by steveepreston** [reply]
@KaKi87 and @yshavit you both talking true.
There is no progress on this feature, and here we are only getting upvote comment notifications

**Reply by adaltonreis** [reply]
> Heh, only meant to downvote for one. I'll retract the other.
> 
> The reason for the downvote is that you're sending everyone who's following this issue a message saying you're also interested in it, rather than just upvoting the issue.

Hello! 

Thanks for clarifying however, there is misundertanding. 

  You see, this is a Github Issue. It's not just a "poll" where only the number of votes matter.  As such is is expected to be a discussion, not a ballot casting mechanism.

So it's normal that I have a field to express my opinion and don't you think that every one who is expecting the solution also feels rather happy to receive a message that shows the increasing number of supporters?

Furthermore, anyone who feels annoyed by receiving the messages can just click on the "unsubscribe" link in their email so they stop receiving them. Very simply.

Let us express our opinion, cause this is expected in this platform (Github). :))

Thank you 🙂

**Comment by maheshmanepally** [comment]
I want this feature too

**Comment by steveepreston** [comment]
Guys, Please go and boost `RooCode` for Jetbrains too https://github.com/RooCodeInc/Roo-Code/discussions/2536

**Comment by ldsx163** [comment]
I want this feature too

**Comment by varunrayen** [comment]
+1 for Jetbrains

**Comment by steveepreston** [comment]
Good news:

"RunVSAgent" JetBrains plugin released by @Naituw and @qdaxb: https://github.com/wecode-ai/RunVSAgent

It supports both Cline and RooCode

![screenshot](https://github.com/user-attachments/assets/bd972c9d-cb5e-4854-a724-2c6d758bc1b4)

**Reply by Sangoku** [reply]
Is this cline or what is that?

**Reply by steveepreston** [reply]
@Sangoku This is [RooCode](https://github.com/RooCodeInc/Roo-Code), a fork of Cline

**Reply by Prikalel** [reply]
shit man, ITS ACTUALLY WORKS!!! THANK YOU 💔💔💀👍😆

**Reply by Sangoku** [reply]
Sadly not rly. I almost always get 403 Project `xxxxxxx` does not have access to model `o3` whent rying any of the models from open ai.

**Reply by juliojgd** [reply]
The calls to the LLM worked ok with https://github.com/wecode-ai/RunVSAgent, but when configuring a MCP server via streamable http following the documentation it always failed

**Reply by chiangzi** [reply]
I tried it, and it works perfectly. Awesome!

https://github.com/wecode-ai/RunVSAgent

<img width="1912" height="989" alt="image" src="https://github.com/user-attachments/assets/f0bda1cc-8fd4-4b23-baf9-e7349d7d0880" />

**Reply by juliojgd** [reply]
Yes, it worked for me too. Also the MCP Servers.

**Reply by lfec** [reply]
Now they support cline
https://github.com/wecode-ai/RunVSAgent/releases

**Reply by Akshay314** [reply]
It shows Cline as uninstalled and installing from VSIX is the only option. Can you please share the link/process?

**Reply by tarrade** [reply]
@Akshay314 just copy the VSIX file from the Github release page. Here a direct link from the latest version https://github.com/cline/cline/releases/download/v3.26.1/cline-3.26.1.vsix

**Reply by Akshay314** [reply]
WOW, Thanks, Cline worked for me, super excited to use it. Why is this answer in comments, can someone (rightful owner) please put it as new comment so that it shows up in latest comments.

BTW does anyone know how can we rely on 3P plugins to be assured its not leaking any data?

**Reply by tigerasks** [reply]
I should like to point out the obvious security and privacy risk that extension is posing.
Personally, I would use with caution and only for non-proprietary code.

**Reply by steveepreston** [reply]
@tigerasks You pointed out the wrong thing. The extension is open-source and also network requests can watched.
The thing you should point is on the api server, where stores pure data.

**Reply by Akshay314** [reply]
@steveepreston So if we are using our private AWS accounts, then there should be no concern wrt privacy/data storage right? Because client side Cline application only call the respective LLM model?

**Reply by isho90** [reply]
@Akshay314  can you please explain how it worked for you in intelliJ, as using my own LLM proxy it always failed because of API request failed and connection failed while with the same API KEY it is correctly working in VS Code

**Reply by Akshay314** [reply]
Not sure about API key, I used AWS credential profile based integration.

Although I would like to call out that my intelliJ was crashing much often when this plugin was enabled, so I had to disable it temporarily.

**Comment by diktamxx** [comment]
+1 for Jetbrains

**Comment by vladP123ggg** [comment]
+1 for Jetbrains

**Comment by danielzayas** [comment]
+1

**Comment by fangzc1** [comment]
+1

**Comment by jiajian-wu** [comment]
It seems to be out: https://plugins.jetbrains.com/plugin/28247-cline/versions/stable

**Reply by steveepreston** [reply]
Finally It's happening!

Edit:
```
Failed to start core

Resource not found:\standalone.zip
java.lang.IllegalStateException: Resource not found:\standalone.zip
at bot.cline.intellij.InstallManager.extractFileFromJar(InstallManager.kt:89)
at bot.cline.intellij.InstallManager.installCoreFromJar(InstallManager.kt:52)
at bot.cline.intellij.core.CoreProcessManager.start(CoreProcessManager.kt:72)
at bot.cline.intellij.core.CoreProcessManager$start$1.invokeSuspend(CoreProcessManager.kt)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:98)
at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:613)
at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:1189)
at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:778)
at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:765)
```

**Reply by KaKi87** [reply]
Wow it's amazing how you caught this as a user before any of the maintainers would make an announcement

**Reply by jsonza6** [reply]
Is this really an official plugin?    Why is there no official announcement

**Reply by KaKi87** [reply]
<img width="298" height="121" src="https://github.com/user-attachments/assets/2ce78a6f-8000-4318-8ccf-639c26bbf51e" />

I would find it weird if JetBrains wouldn't check email addresses.

**Reply by scottbot95** [reply]
I get the same "failed to start" error. Seems pretty likely to me that this is not actually meant for public consumption yet given there was no announcement, the version is 0.0.1, and it seemingly doens't actually even work.

**Reply by steveepreston** [reply]
@scottbot95 The strange thing is that in the JetBrains Store Page there is many 5-star comments, while it's even not load!

**Reply by KaKi87** [reply]
I can see the reviews.

<img width="802" height="1540" alt="screenshot-plugins-jetbrains-com-2025-08-31-20-57-57" src="https://github.com/user-attachments/assets/52c775c3-3426-4a12-ba80-399b7d700718" />

**Reply by dorgrn** [reply]
> I can see the reviews.
> 
> <img alt="screenshot-plugins-jetbrains-com-2025-08-31-20-57-57" width="802" height="1540" src="https://private-user-images.githubusercontent.com/21284089/483989155-52c775c3-3426-4a12-ba80-399b7d700718.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY3NTQwOTEsIm5iZiI6MTc1Njc1Mzc5MSwicGF0aCI6Ii8yMTI4NDA4OS80ODM5ODkxNTUtNTJjNzc1YzMtMzQyNi00YTEyLWJhODAtMzk5YjdkNzAwNzE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTAxVDE5MDk1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU1MDFiNDdkNTBmYTg2YzE3NDNhYjRhYjlkMGEyZDllZTNlZDUwMDQ0OTQ3N2IzMjViMDVhMGMwMTRhMDdjMmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RgrUe7lfmz4J8oD1OuXU40z7DYCp53dt9LMeKjXZiTU">

These are mostly cline team members (and maybe f&f)

**Reply by steveepreston** [reply]
@dorgrn It's fine, we just need Cline there

**Comment by zerubeus** [comment]
It is open source ?

**Reply by KaKi87** [reply]
Oh, that's a good question.

I assumed both extensions would be on the same repo, but maybe I shouldn't have :sweat_smile:

**Reply by rmonvfer** [reply]
Bumping this, I've been looking for the source somewhere but haven't been able to find it anywhere... it doesn't seem to be on another branch or in other repo in the cline organization, which is weird.

**Reply by ehartford** [reply]
I would like to make some modifications and contributions - but I can't seem to find the code - is it open source?

**Comment by jclab-joseph** [comment]
Documentation has also been added: https://docs.cline.bot/getting-started/installing-cline-jetbrains

And today is version 0.0.2, and the basic functionality seems to be working.

**Reply by linuxfreak** [reply]
Thank you

**Reply by timkindberg** [reply]
When I used Cline in VS Code I selected the "VS Code LM API" Provider which allowed me to then select from the Copilot models. The reason is that Copilot is approved by our security dept at work. But that same option isn't working in JetBrains, even though I have the Copilot extension installed as well in that IDE.

Note how I can select the model after selecting "VS Code LM API"
<img width="592" height="447" alt="image" src="https://github.com/user-attachments/assets/8a6e7fb9-fe49-4cca-bb59-add82b4a2987" />

Note how in PyCharm I cannot, I only get the single dropdown.
<img width="783" height="438" alt="image" src="https://github.com/user-attachments/assets/ddad2100-e9df-46c9-b7ef-06a2b41e4768" />

I would need this to work before I could use Cline in PyCharm.

**Reply by leonzy** [reply]
> Documentation has also been added: https://docs.cline.bot/getting-started/installing-cline-jetbrains
> 
> And today is version 0.0.2, and the basic functionality seems to be working.

where is the source code for this plugin? 
Will it be open sourced later?

**Comment by marek1990kaminski** [comment]
bump for IntelliJ

**Comment by Prikalel** [comment]
0.0.2 Not working for rider
--
Отправлено из  Mail для Android вторник, 16 сентября 2025г., 10:46 +03:00 от 张勇  ***@***.*** :

**Comment by fracasula** [comment]
The font size is huge, it seems to be ignoring the `Editor` →`Font` settings.

Here's how Cline looks like:

<img width="1341" height="2059" alt="image" src="https://github.com/user-attachments/assets/60fbbc6c-3791-4fd1-97cf-1ed644fbd424" />

Vs another window in the same editor:

<img width="1341" height="2059" alt="image" src="https://github.com/user-attachments/assets/7cf94fe3-f2e6-4b3d-9025-3da2f7c5ae75" />

**Comment by isho90** [comment]
Did anyone tried cline on intellij with LiteLLM ? it is always not working 

<img width="610" height="1025" alt="image" src="https://github.com/user-attachments/assets/8848ac5d-79ea-45ed-ba9a-d821057cd5cf" />

Same config and API Key is working in VS code

**Reply by juliojgd** [reply]
It's working for me with LiteLLM. Environment:
- WSL2
- IntelliJ 2025.2
- Cline plugin for IntelliJ 1.0.0 (That's Cline 3.28.0)) 

But:
Terminal does not work at all. It tries to run the terminal commands embedded in the task conversation, it does not send the command to the terminal window. See:
<img width="1230" height="794" alt="image" src="https://github.com/user-attachments/assets/47555e39-7179-44c2-a162-144135a04ab0" />

It cannot find the Maven executable at all. But in the terminal opened inside IntelliJ does work flawlessly:
<img width="1307" height="374" alt="image" src="https://github.com/user-attachments/assets/5e9ed553-9a90-42b7-8bd2-1d42607a5238" />

Neither the default or bash terminal settings work. See here:
<img width="1062" height="534" alt="image" src="https://github.com/user-attachments/assets/96bfc0cf-948e-426f-8866-a67c053fcdd6" />

(1) - Remains a reference to VSCode
(2) and (3) This help links opened a blank window that does not load anything.

Without terminal working is useless for most use cases.

**Reply by juliojgd** [reply]
@csells Maybe you were trying to reply to my post? The environment seems to be fine as in a terminal inside IntelliJ everything is working. If Cline sent the commands to the terminal I think it will work, but it tries to run itsenf inside the task concersation

**Reply by isho90** [reply]
@juliojgd it is strange I also checked the communication layer but always getting API request error 
Connection failed.

**Comment by csells** [comment]
@sjf it seems like there may be an environment issue as the root cause of the problems with the terminal and maven.

**Comment by nickbaumann98** [comment]
Hey everyone -- we just went GA with Cline for JetBrains. Here's the link to install: https://plugins.jetbrains.com/plugin/28247-cline

Thank you for your patience ❤️

**Reply by maxious** [reply]
@nickbaumann98 is the source code available?

**Reply by Inrego** [reply]
Hi - on your website you list Rider as supported:
<img width="1195" height="622" alt="image" src="https://github.com/user-attachments/assets/bec72f29-b593-4aee-8ef7-87b17ab35d68" />

But in reality, it seems to not support Rider:
<img width="1233" height="207" alt="image" src="https://github.com/user-attachments/assets/00677bee-67be-4c10-8280-a7c873112ff9" />

**Reply by nickbaumann98** [reply]
Hey there -- we just shipped a patch for this yesterday and it will be going out in the next release. We heard from a LOT of users who were looking to use Rider.

**Reply by KaKi87** [reply]
@nickbaumann98 is the source code available????

**Reply by ehartford** [reply]
I am looking for the source code - is it open source? @nickbaumann98

**Reply by reza-repo** [reply]
<img width="340" height="427" alt="image" src="https://github.com/user-attachments/assets/7226d774-03b8-4a6c-a433-17e821d74114" />

Lets go button cannot be clicked with `VS Code LM API`

this is the error I see in the logs:

#bot.cline.server.ts ProtoBus request: /cline.ModelsService/getVsCodeLmModels
Error fetching VS Code LM models: TypeError: Cannot read properties of undefined (reading 'selectChatModels')
    at getVsCodeLmModels

**Comment by wzlstudy** [comment]
<img width="585" height="612" alt="image" src="https://github.com/user-attachments/assets/6d57123a-05d6-44b9-9186-42da6f4ea19b" />

There are the following issues on the IDEA:

1. The checkboxes do not display;
2. Can an option be provided to control whether it is fixed on the left/right side?

**Reply by ngdriss** [reply]
which version you have ? I tried with various version on Ultimate but not being able to display this view.

**Reply by nickbaumann98** [reply]
You can drag the Cline icon to the right or left sidebar!

**Reply by wzlstudy** [reply]
IntelliJ IDEA 2025.2.1 (Ultimate Edition) and macOS 12.7.6 and Cline 1.0.0

**Reply by wzlstudy** [reply]
> You can drag the Cline icon to the right or left sidebar!

Yes, but every time I open a new project, it is displayed on the left by default and I need to adjust it manually. I'm thinking if a global configuration could be provided so that I don't have to adjust it manually every time I open a new project

**Reply by sxnamit** [reply]
> which version you have ? I tried with various version on Ultimate but not being able to display this view.

Were you able to resolve this? Cline tool window remains empty for me too. I am on IntelliJ IDEA 2025.2.2 (Ultimate Edition), Cline 1.0.7 (Stable), macOS Tahoe 26.0.
Checked the cline logs, idea logs...didn't see any error/exception there either.

**Comment by l-wieczorek** [comment]
When can we expect working integration with Github Copilot (VS Code LM API)?

**Reply by timkindberg** [reply]
I opened a bug for this: https://github.com/cline/cline/issues/6077

**Comment by Fuud** [comment]
Hi!

Could you please opensource to fork and PR plugin repo https://github.com/cline/intellij-plugin/. Currently it is private.

**Reply by jWeck-Software** [reply]
> Hi!
> 
> Could you please opensource to fork and PR plugin repo https://github.com/cline/intellij-plugin/. Currently it is private.

This will not happen, since they are charging $200/u/m "anual payment" for it as of 2026.
https://cline.bot/pricing
To get the Teams-License, you have to buy at least 10 seats.

None organisations, freelancers and individuals are not able to buy the license at all and use the plugin.

It is much more expensive as advertised on the website!
Not $200, it is $240.

<img width="453" height="433" alt="Screenshot_20251030_104552" src="https://github.com/user-attachments/assets/9ac1d734-ead8-4473-9c86-eedb3028eb67" />

**Reply by ehartford** [reply]
Roo code has open source jetbrains plug-in

https://github.com/RooCodeInc/Roo-Code-JetBrains
